src/classes/CW_CampaignTriggerHandlerTest.cls

```diff
@@ -0,0 +1,266 @@
+@isTest
+private class CW_CampaignTriggerHandlerTest {
+
+/**
+* File Name   :    CW_CampaignTriggerHandlerTest.cls
+* Description :    Test Class for CW_CampaignTriggerHandler.cls
+*
+* Modification Log
+* ============================================================================
+* Ver Date       Author         Modification
+* --- ---------- -------------- --------------------------
+* 0.1 29/04/2014 Chris Yorkston	Initial version
+* 0.2 06/06/2014 Klea Kolaric	D-35748
+*
+*
+*/
+	// test that points on Campaign (child) are rolled up to Roadmap (parent) when Campaign records are created, updated, deleted and undeleted
+	@isTest static void testCampaignRollupSumOnRoadmap() {
+		
+		// setup data
+		CW_Roadmap__c roadmap = TestDataUtil.getRoadmap();
+		insert roadmap;
+		
+		List<CW_Campaign__c> camps = TestDataUtil.getCampaigns(2);
+		for (CW_Campaign__c c :camps) {
+			c.Roadmap__c = roadmap.Id;
+			c.Points__c = 5.0;
+		}
+		
+		// insert Campaign (with points)
+		insert camps;
+		
+		// assert inserted campaigns (child) points are rolled up to roadmap (parent)
+		List<CW_Roadmap__c> roadmaps = [select Id, Points_Used__c from CW_Roadmap__c where Id = :roadmap.Id];
+		system.assertEquals(10.0, roadmaps[0].Points_Used__c);
+		
+		// update Campaigns (with points)
+		camps[0].Points__c = 20.0;
+		camps[1].Points__c = 40.0;
+		update camps;
+		
+		// assert updated campaigns (child) points are rolled up to roadmap (parent)
+		roadmaps = [select Id, Points_Used__c from CW_Roadmap__c where Id = :roadmap.Id];
+		system.assertEquals(60.0, roadmaps[0].Points_Used__c);
+		
+		// delete Campaign
+		delete camps[0];
+		
+		// assert deleted campaign (child) points are not rolled up to roadmap (parent)
+		roadmaps = [select Id, Points_Used__c from CW_Roadmap__c where Id = :roadmap.Id];
+		system.assertEquals(40.0, roadmaps[0].Points_Used__c);
+		
+		// undelete Campaign
+		undelete camps[0];
+		
+		// assert undeleted campaign (child) points are included in roll up to roadmap (parent)
+		roadmaps = [select Id, Points_Used__c from CW_Roadmap__c where Id = :roadmap.Id];
+		system.assertEquals(60.0, roadmaps[0].Points_Used__c);
+		
+	}
+	
+	// test that points on Campaign (child) are rolled up to Roadmap (parent) when lookup to Roadmap is removed from a Campaign record
+	@isTest static void testCampaignRollupSumOnRoadmapWhenLookupToRoadmapRemoved() {
+		
+		// setup data
+		CW_Roadmap__c roadmap = TestDataUtil.getRoadmap();
+		insert roadmap;
+		
+		List<CW_Campaign__c> camps = TestDataUtil.getCampaigns(2);
+		for (CW_Campaign__c c :camps) {
+			c.Roadmap__c = roadmap.Id;
+			c.Points__c = 5.0;
+		}
+		
+		// insert Campaign (with points)
+		insert camps;
+		
+		// assert inserted campaigns (child) points are rolled up to roadmap (parent)
+		List<CW_Roadmap__c> roadmaps = [select Id, Points_Used__c from CW_Roadmap__c where Id = :roadmap.Id];
```

Comment:
Problem: Unbound SOQL query. Reason: a SOQL query can return up to 50,000 records. If this number is exceeded a runtime exception is thrown. Suggestion: cap the max number of records returned adding a LIMIT clause to the query.